docs/design/simplified_federation_deployment.md

```diff
@@ -0,0 +1,388 @@
+3. Runs:
+
+    ```shell
+    $ kubeadm init federation
+    ```
+
+   to set up a Kubernetes control plane and a federation control plane
+   on that machine.
+
+
+#### Cluster Registration Experience
+
+1. User downloads the
+   [`kubectl`](http://kubernetes.io/docs/getting-started-guides/kubeadm/)
+   executable on their local computer, i.e. the computer they use for
+   their day-to-day activities.
+2. Runs:
+
+    ```shell
+    $ kubectl register mycluster
+    ```
+
+   to register their cluster named `mycluster`, where `mycluster` is the
+   name of the cluster's context in the local kubeconfig.
+
+#### Cluster Deregistration Experience
+
+1. To deregister a cluster, user runs:
+
+    ```shell
+    $ kubectl deregister mycluster
+    ```
+
+### Points for Discussion:
+
+1. Is `unregister` a better verb than `deregister`?
+2. Does the verb pair `join/unjoin` make more sense in this context?
+
+
+## Design
+
+
+#### Federation Control Plane Bootstrap
+
+`kubeadm init` initializes a Kubernetes control plane. In order to do
+that, it generates the manifests for the various control plane
+components and starts them. It also generates the certificates and the
+credentials required for those components to interact with the API
+server.
+
+Federation control plane deployment builds on top of this functionality.
+Running `kubeadm init federation` first performs all the operations
+involved in bootstrapping a Kubernetes control plane, as running
+`kubeadm init` would. We henceforth call this Kubernetes cluster for
+which the control plane was initialized, the "bootstrap cluster". Upon
+initializing the bootstrap cluster, we perform the following operations
+to bootstrap a federation control plane.
+
+
+**1. Create a namespace for federation system components**
+
+* Call this namespace `federation-system` to keep it consistent with
+  Kubernetes.
+* The namespace is created in the bootstrap cluster.
+
+**2. Expose a network endpoint for the federation API server**
+
+* Create a "Loadbalancer" type Kubernetes service in the
+  `federation-system` namespace in the bootstrap cluster to expose the
+  yet to be created federation API server.
+* Wait until a load balancer IP address is allocated to the service.
+
+**3. Generate TLS certificates and credentials**
+
+* Using `kubeadm`'s PKI infrastructure, generate TLS certificates for a
+  new CA, yet to be created federation API server and federation
+  controller manager.
+* Generate credentials for the federation API server.
+
+**4. Create a kubeconfig secret**
+
+* Using the load balancer IP address of the federation API server and
+  the generated certificates and credentials, generate a kubeconfig
+  file for the federation controller manager.
+* Create a secret in the bootstrap cluster's `federation-system`
+  namespace and populate it with the contents of the generated
+  kubeconfig file.
+
+**5. Create a persistent volume and a claim to store the federation API server's state**
```

Comment:
The advantage of using a PD is that we can tear down the federation (all the members of the quorum) and still keep its state. And we could resume from that state later. PD here is not for HA, but for pause and resume. Or simply persistence.